homeassistant/components/iotty/cover.py

```diff
@@ -0,0 +1,208 @@
+
+from . import IottyConfigEntry
+from .api import IottyProxy
+from .const import DOMAIN
+from .coordinator import IottyDataUpdateCoordinator
+from .entity import IottyEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: IottyConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Activate the iotty Shutter component."""
+    _LOGGER.debug("Setup COVER entry id is %s", config_entry.entry_id)
+
+    coordinator = config_entry.runtime_data.coordinator
+    entities = [
+        IottyShutter(
+            coordinator=coordinator, iotty_cloud=coordinator.iotty, iotty_device=d
+        )
+        for d in coordinator.data.devices
+        if d.device_type == SH_DEVICE_TYPE_UID
+        if (isinstance(d, Shutter))
+    ]
+    _LOGGER.debug("Found %d Shutters", len(entities))
+
+    async_add_entities(entities)
+
+    known_devices: set = config_entry.runtime_data.known_devices
+    for known_device in coordinator.data.devices:
+        if known_device.device_type == SH_DEVICE_TYPE_UID:
+            known_devices.add(known_device)
+
+    @callback
+    def async_update_data() -> None:
+        """Handle updated data from the API endpoint."""
+        if not coordinator.last_update_success:
+            return
+
+        devices = coordinator.data.devices
+        entities = []
+        known_devices: set = config_entry.runtime_data.known_devices
+
+        # Add entities for devices which we've not yet seen
+        for device in devices:
+            if (
+                any(d.device_id == device.device_id for d in known_devices)
+                or device.device_type != SH_DEVICE_TYPE_UID
+            ):
+                continue
+
+            iotty_entity = IottyShutter(
+                coordinator=coordinator,
+                iotty_cloud=coordinator.iotty,
+                iotty_device=Shutter(
+                    device.device_id,
+                    device.serial_number,
+                    device.device_type,
+                    device.device_name,
+                ),
+            )
+
+            entities.extend([iotty_entity])
+            known_devices.add(device)
+
+        async_add_entities(entities)
+
+    # Add a subscriber to the coordinator to discover new devices
+    coordinator.async_add_listener(async_update_data)
+
+
+class IottyShutter(IottyEntity, CoverEntity):
+    """Haas entity class for iotty Shutter."""
+
+    _attr_device_class = CoverDeviceClass.SHUTTER
+    _iotty_device: Shutter
+
+    def __init__(
+        self,
+        coordinator: IottyDataUpdateCoordinator,
+        iotty_cloud: IottyProxy,
+        iotty_device: Shutter,
+    ) -> None:
+        """Initialize the Shutter device."""
+        super().__init__(
+            coordinator=coordinator, iotty_cloud=iotty_cloud, iotty_device=iotty_device
+        )
+
+        self._iotty_device = iotty_device
```

Comment:
This one as well

Reply:
The issue with this one is that I want it to be a specific subtype (Shutter, here), so if I move it to the shared base entity I will only have a "Device" and will need to cast it every time, having to make extra checks every time

Reply:
Nope, you already overwrite the type here, so when you reference `self._iotty_device` in this class it's always seen as Shutter

Reply:
Neat, I made this change